### ISSUE TITLE
J3: repeatable Subform remove & add

### ISSUE THREAD

**Issue by mggcodes** [joomla/joomla-cms#27435]
### Steps to reproduce the issue
- login to administrator 
- edit an existing item with a repeatable subform (the subform should have more then 1 entry already)
- delete the first entry of the repeatable
- add a new entry to the repeatable 
- save the item

### Expected result
at the function JModelAdmin::save($data)

all entries of the subform should be in the $data object

```
$data['subform']['1']
$data['subform']['2']

```
example code in Editform:
```
<div class="subform-repeatable-group subform-repeatable-group-sr-0" data-base-name="eventdates" data-group="eventdates1">
<div class="subform-repeatable-group subform-repeatable-group-sr-0" data-base-name="eventdates" data-group="eventdates2" data-new="true">
```
### Actual result
at the function JModelAdmin::save($data)

the entry you added get the same id like the existing entry. So the new one will override the existing one.

`$data['subform']['1']`

example code in Editform:
```
<div class="subform-repeatable-group subform-repeatable-group-sr-0" data-base-name="eventdates" data-group="eventdates1">
<div class="subform-repeatable-group subform-repeatable-group-sr-0" data-base-name="eventdates" data-group="eventdates1" data-new="true">
```

### System information (as much as possible)
Joomla Version 3.9.X (tested on 3.9.12 and 3.9.14)

**Comment by richard67** [comment]
Should be already solved with PR #27334 . @mggcodes Please check, and if solved close this issue.

**Comment by Fedik** [comment]
yeah, that the same as #27307 
the fix (#27334) will be available in 3.9.15

**Comment by alikon** [comment]
we already have a fix and will be available in 3.9.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27435">issues.joomla.org/tracker/joomla-cms/27435</a>.</sub>
